src/Compilers/Core/Rebuild/CompilationOptionsReader.cs

```diff
@@ -155,6 +152,11 @@ public ImmutableArray<MetadataReferenceInfo> GetMetadataReferences()
             var mdReader = PeReader.GetMetadataReader();
             var methodDefinition = mdReader.GetMethodDefinition(header.EntryPoint);
             var methodName = mdReader.GetString(methodDefinition.Name);
+            if (methodName != WellKnownMemberNames.EntryPointMethodName)
```

Comment:
Why return `null` here vs. adding an extra field to the tuple that says `IsTopLevelStatements`? 

The current approach is in some ways hiding information and it's asking callers to infer behavior based on that information missing. Callers have to know that when this is `null` then "ah it's top level statements, do another trick". Rather than that why not return all the info and explicitly state this is top level statements case?

Reply:
In my view, the purpose of the method is to answer the question "what main type name should I pass to the compilation/emit options." If the answer is "none", then it's ok to return null. Would be happy to rename this to try and make that more explicit.

Reply:
Here is another way of phrasing my question: today `null` can represent two different outcomes

1. Inability to read the metadata correctly 
2. Metadata read correctly but this is a program compiled with top level statements

Should consumers react the same way to both situations?

Reply:
> 1. Inability to read the metadata correctly

I think if the tool can detect that the metadata is *malformed*, it might be good to throw and turn this into a misc error. But if the metadata is just absent, it feels ok for the consumer to do what they were doing before this PR--which is to react to the null by saying "ok, I just won't pass anything for the mainType or the debugEntryPoint."